### PR TITLE
chore(build-system-test): update web framework and build tool versions, parameterize options

### PR DESCRIPTION
**Pull request by esauerbo**
#### Description of changes
- Remove `cra` (deprecated) and all references to it, set new default build tool to `next`
- Replace `next` 12 and `vite` 2 with 13 and 3, the current lowest-supported versions 
- Remove the `<framework>-<version>-<buildtool-version>...` scripts from the build tests in favor of                                
 `test:<framework>:<build-tool>` with other options included as flags directly in the reusable-build-test script, which makes it less cumbersome to add new versions/tags/etc 
- Remove some unused options and templates 

#### Description of how you validated changes

Passing POC workflow that triggers web build tests https://github.com/aws-amplify/amplify-ui/pull/5869 

#### Checklist

- [x] Have read the [Pull Request Guidelines](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md)
- [x] PR description included
- [x] `yarn test` passes and tests are updated/added
- [x] PR title and commit messages follow [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/#summary) syntax

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: build-system-tests/scripts/setup-mega-app.sh [hunk 1]
```diff
@@ -33,10 +31,6 @@ while [[ $# -gt 0 ]]; do
     BUILD_TOOL_VERSION=$2
     shift
     ;;
-  -l | --language)
```
Comment (esauerbo) [inline build-system-tests/scripts/setup-mega-app.sh:36]: This option was only being used for `cra` so pulling it out altogether

File: .github/workflows/reusable-build-system-test.yml [hunk 2]
```diff
@@ -17,53 +17,24 @@ jobs:
       matrix:
         framework: [react]
         framework-version: [latest]
-        build-tool: [cra, next, vite]
+        build-tool: [next, vite]
         build-tool-version: [latest]
         pkg-manager: [npm]
         language: [ts]
-        node-version: [18]
+        node-version: [20]
```
Comment (reesscot) [inline .github/workflows/reusable-build-system-test.yml:24]: nice!

File: build-system-tests/scripts/mega-app-copy-files.sh [hunk 3]
```diff
@@ -110,15 +81,6 @@ if [[ "$FRAMEWORK" == 'react' && "$BUILD_TOOL" == 'vite' ]]; then
     cp $AWS_EXPORTS_DECLARATION_FILE mega-apps/${MEGA_APP_NAME}/src/aws-exports.d.ts
     echo "cp templates/components/react/vite/App.tsx mega-apps/${MEGA_APP_NAME}/src/App.tsx"
     cp templates/components/react/vite/App.tsx mega-apps/${MEGA_APP_NAME}/src/App.tsx
-
-    # See troubleshooting:
-    # https://ui.docs.amplify.aws/react/getting-started/troubleshooting#vite
```
[the remark this answers is not in the document]
Comment (reesscot) [reply build-system-tests/scripts/mega-app-copy-files.sh:115]: Good find!

**Review by reesscot — APPROVED** [review]
Changes look great, thanks for the cleanup!

**Review by dbanksdesign — APPROVED** [review]

**Pull request merged**